packages/custom_lint/lib/src/v2/server_to_client_channel.dart

```diff
@@ -90,33 +104,50 @@ void _writePackageConfigForTempProject(
         continue;
       }
 
-      final currentPackage = packageMap[package.name];
+      final currentPackageWithContextRoot = packageMap[package.name];
+      final currentPackage = currentPackageWithContextRoot?.package;
 
       if (currentPackage != null && currentPackage.root != package.root) {
-        throw StateError(
-          '''
-Two ContextRoots depend on ${package.name} but use different version,
-therefore custom_lint does not know which one to pick.
-- ${package.root}
-- ${currentPackage.root}
-''',
+        packagesInMultipleContextRoots.add(
+          _PackageAndContextRootPair(
+            currentPackageWithContextRoot!,
+            _PackageAndContextRoot(package, contextRoot),
+          ),
         );
       }
 
-      packageMap[package.name] = package;
+      packageMap[package.name] = _PackageAndContextRoot(package, contextRoot);
     }
   }
+  if (packagesInMultipleContextRoots.isNotEmpty) {
+    throw StateError(
+      'The following packages are included in multiple ContextRoots:\n'
+      '${[
+        for (final pair in packagesInMultipleContextRoots)
+          '- ${pair.first.package.name}\n'
```

Comment:
We'd want to show the relative path to the projects too.

Reply:
isn't `pair.first.contextRoot.root` doing this. It prints the absolute path of the package, like:

`/Users/adson.leal/dev/sora/sora-world/sora_app/packages/home`

Reply:
It probably should be a relative path though